[17.0][MIG] product_tax_multicompany_default: Migration to version 17.0

carlos-lopez-tecnativa opened the pull request:
@Tecnativa TT49882

NOTE: It is necessary to use `sudo` in test because the `account.group_account_manager` group does not have permission to create products. See https://github.com/odoo/odoo/commit/512574861691f425ec6a17f20fe4b586bb88a299#r144593878

carlosdauden committed:
[9.0][WIP] product_tax_multicompany_default: New module

pedrobaeza committed:
[FIX] product_tax_multicompany_default: Fix tests + don't overwrite self

OCA Transbot updated translations from Transifex

cubells committed:
[MIG] product_tax_multicompany_default: Migration to 11.0

[UPD] Update product_tax_multicompany_default.pot

[UPD] Update product_tax_multicompany_default.pot

Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

[ADD] icon.png

carlosdauden committed:
[MIG] product_tax_multicompany_default: Migration to 12.0

Commit:
[MIG] product_tax_multicompany_default: Migration to 13.0

celm1990 committed:
[MIG] product_tax_multicompany_default: Migration to 15.0
* Use TransactionCase instead of SavepointCase
* adapt create method for multi records

Shide committed:
[MIG] product_tax_multicompany_default: Migration to 16.0

Shide committed:
[FIX] product_tax_multicompany_default: Overwrite taxes instead of adding it

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-16.0/multi-company-16.0-product_tax_multicompany_default
Translate-URL: https://translation.odoo-community.org/projects/multi-company-16-0/multi-company-16-0-product_tax_multicompany_default/

Shide committed:
[FIX] product_tax_multicompany_default: Improve test coverage

- Refactored some tests to use new_test_user and @users
- Added a new test to cover the workflow of the product Form

chafique-delli committed:
[16.0][FIX] product_tax_multicompany_default: supplier_taxes_id was not found in the form view during tests

yajo committed:
[IMP] product_tax_multicompany_default: beautier and mass propagation

- Propagate button was ugly. It's prettier now.
- Add ability to filter products that have divergent cross-company taxes.
- Allow propagating massively.

@moduon MT-2586

yajo committed:
[ADD] product_account_multicompany_default: propagate product account to other companies

This module extends the functionality of products and accounting to allow you
to propagate multi-company product accounts from the current company to all
other companies where you have access rights.

@moduon MT-2626

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: multi-company-16.0/multi-company-16.0-product_tax_multicompany_default
Translate-URL: https://translation.odoo-community.org/projects/multi-company-16-0/multi-company-16-0-product_tax_multicompany_default/it/

carlos-lopez-tecnativa committed:
[IMP] product_tax_multicompany_default: pre-commit stuff

carlos-lopez-tecnativa committed:
[MIG] product_tax_multicompany_default: Migration to 17.0
NOTE: It is necessary to use sudo in test because the account.group_account_manager group does not have permission to create products.
See https://github.com/odoo/odoo/commit/512574861691f425ec6a17f20fe4b586bb88a299#r144593878

carolinafernandez-tecnativa commented:
I have tested and check  this example:

1- I have 2 companies
2- Select with user two companies
3- Create new product selecting two companies with user.

Taxes for first company is added twice
<img width="1215" alt="image" src="https://github.com/user-attachments/assets/bd44fe2a-cb00-4585-a8e0-78e15a7aaf89">

If i edit product then it is fixed.

Is it okay this behaviour?

carlos-lopez-tecnativa committed:
[FIX] product_tax_multicompany_default: Filter taxes according to the expected scope.

 When a user has multiple companies selected and multiple taxes with the same name, searching can return more taxes outside the intended scope (sale or purchase). This commit adds a new explicit filter to type_tax_use to avoid retrieving incorrect taxes.

carlos-lopez-tecnativa commented:
@carolinafernandez-tecnativa could you please test again?, thanks

carolinafernandez-tecnativa reviewed (approved):
Functional review LGTM :) Thanks!

pedrobaeza commented:
/ocabot migration product_tax_multicompany_default

pedrobaeza commented on product_tax_multicompany_default/tests/test_product_tax_multicompany.py:
```diff
@@ -101,27 +120,35 @@ def test_multicompany_default_tax(self):
-            }
+        product = (
+            self.env["product.product"]
+            .sudo()
```
Why do you need the sudo? It shouldn't be needed.

carlos-lopez-tecnativa replied:
@pedrobaeza In the PR description, I justified these changes. Basically, the `account.group_account_manager` group does not have permission to create products.

I also added comments in the code to clarify and removed some unnecessary `sudo()` calls after create. Let me know if you have any other comments. Thank you.

pedrobaeza reviewed (approved):
/ocabot merge nobump
